validator/client/src/main/java/tech/pegasys/teku/validator/client/ValidatorClientService.java

```diff
@@ -192,15 +198,65 @@ public static ValidatorClientService create(
             validatorRegistrator,
             config.getSpec(),
             services.getMetricsSystem());
+
+    validatorClientService.initializeValidators(validatorApiChannel, asyncRunner);
+
+    SafeFuture<Void> validatorsInitialized = new SafeFuture<>();
+
     asyncRunner
         .runAsync(
-            () ->
-                validatorClientService.initializeValidators(
-                    config, validatorApiChannel, asyncRunner))
-        .propagateTo(validatorClientService.initializationComplete);
+            () -> validatorClientService.initializeValidators(validatorApiChannel, asyncRunner))
+        .propagateTo(validatorsInitialized);
+
+    validatorsInitialized
+        .thenCompose(
+            __ -> {
```

Comment:
at the end here you can simply do:
<img width="785" alt="image" src="https://user-images.githubusercontent.com/15999009/203027933-5cc2b224-e78f-49e5-8607-939d98ba19f4.png">

Reply:
Good catch, since the `runAsync` already returns a `SafeFuture` that would be cleaner...
Done